### DISCUSSION TITLE
Can't get full calendar to work

### DISCUSSION THREAD

**Discussion by astroman68** [Alexays/Waybar#2145] [Q&A] [answered]
I am trying to get the full calendar to display as the tool tip for the clock module. I am using the example from the waybar [wiki](https://github.com/Alexays/Waybar/wiki/Module:-Clock) e.g. 

```
"clock": {
        "format": "{:%H:%M}  ",
        "format-alt": "{:%A, %B %d, %Y (%R)}  ",
        "tooltip-format": "<tt><small>{calendar}</small></tt>",
        "calendar": {
                    "mode"          : "year",
                    "mode-mon-col"  : 3,
                    "weeks-pos"     : "right",
                    "on-scroll"     : 1,
                    "on-click-right": "mode",
                    "format": {
                              "months":     "<span color='#ffead3'><b>{}</b></span>",
                              "days":       "<span color='#ecc6d9'><b>{}</b></span>",
                              "weeks":      "<span color='#99ffdd'><b>W{}</b></span>",
                              "weekdays":   "<span color='#ffcc66'><b>{}</b></span>",
                              "today":      "<span color='#ff6699'><b><u>{}</u></b></span>"
                              }
                    },
        "actions":  {
                    "on-click-right": "mode",
                    "on-click-forward": "tz_up",
                    "on-click-backward": "tz_down",
                    "on-scroll-up": "shift_up",
                    "on-scroll-down": "shift_down"
                    }
    },
```
I have played around with the properties off and on for a couple of days now and no matter what I do I continue to get the standard month calendar. I have also scoured the internet looking for some dotfiles that might contain something not shown in the wiki. After reviewing many, most seem to either not use the calendar at all, or just the monthly.

One caveat, I am using waybar-hyrland, but I have also tried removing that and installing stock waybar and it does the same thing. Any help much appreciated.

**Comment by LukashonakV** [comment, marked as answer]
Hi @astroman68 , according [Waybar WIKI Clock](https://github.com/Alexays/Waybar/wiki/Module:-Clock)
> The clock module displays the current date and time.
>
>Note: There are two clock implementations:
>
>    clock: Fully consistent with what is written below. This is enabled if either one of the following build time conditions is met:
>
>        C++20 with __cpp_concepts >= 201907 (gcc >=13)
>        [date.h](https://github.com/HowardHinnant/date) is installed
>
>    simpleclock: Provides only date time functionality. That's it. Conditions for using: False the first point.

I guess it might be exact your issue and waybar is built with simpleclock module only. 
In build log you should check something like this:
`Checking if "nl_langinfo with _NL_TIME_WEEK_1STDAY, _NL_TIME_FIRST_WEEKDAY" : links: YES`
In case when it says: `: links: NO` it means waybar will use simpleclock instead of usual clock module

---
upd: please also check in build log something like this: `Computing int of "__cpp_lib_chrono" : 201611` . What the value in your case ?

**Comment by astroman68** [comment]
Thanks for the help. I guess I misunderstood what that portion of the wiki was saying. I took it to mean there were two 'options' to the clock module. One being '#clock' that would give you the full functionality or '#simpleclock' that would just give you date/time. In hindsight, I guess it doesn't make sense to take it that way. Anyway, I am sure you are right that I am missing that header (date.h) if it is not included in the binary build in the AUR. Now that I have a better understanding of how it works, I will work some more with it and see if I can get it working.

**Comment by teenw0** [comment]
The order of days starts from Sunday, how and where can it be changed?

**Reply by izmyname** [reply]
You can add  `"locale": "en_GB.UTF-8"` to your waybar clock module config or export `LC_TIME=en_GB.UTF-8 ` variable to your environment.

The variable might be preferable, as it also fixes the output of `cal` and similar commands.

en_GB is a British English locale
